### PR TITLE
[SPARK-40836][CONNECT] AnalyzeResult should use struct for schema

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

This PR replace column names and columns type with a schema (which is a struct).

### Why are the changes needed?

Before this PR, AnalyzeResult separates column names and column types. However these two can be combined to form a schema which is a struct.  This PR will simplify that proto message.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @grundprinzip @hvanhovell

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala [hunk 1]
```diff
@@ -50,11 +50,27 @@ object DataTypeProtoConverter {
         proto.DataType.newBuilder().setI32(proto.DataType.I32.getDefaultInstance).build()
       case StringType =>
         proto.DataType.newBuilder().setString(proto.DataType.String.getDefaultInstance).build()
+      case LongType =>
```
Comment (grundprinzip) [inline connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala:53]: nit: for readability, move this to the other int type?

**Comment by amaliujia** [comment]
R: @cloud-fan

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala [hunk 2]
```diff
@@ -50,11 +50,27 @@ object DataTypeProtoConverter {
...
   }
 
+  def toConnectProtoStructType(schema: StructType): proto.DataType = {
+    val struct = proto.DataType.Struct.newBuilder()
+    for (structField <- schema.fields) {
+      struct.addFields(
+        proto.DataType.StructField
+          .newBuilder()
+          .setName(structField.name)
+          .setType(toConnectProtoType(structField.dataType)))
```
Comment (cloud-fan) [inline connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala:69]: we also need to set nullablity.

Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala:69]: ok let me fix the nullability in this PR then. Maybe it is a right time now.

File: python/pyspark/sql/connect/client.py [hunk 3]
```diff
@@ -151,7 +151,44 @@ def _to_pandas(self, plan: pb2.Plan) -> Optional[pandas.DataFrame]:
...
+            for proto_field in schema.struct.fields:
+                structFields.append(
+                    StructField(
+                        proto_field.name,
+                        self._proto_schema_to_pyspark_schema(proto_field.type),
+                        proto_field.nullable,
+                    )
+                )
+            return StructType(structFields)
+        elif schema.HasField("i64"):
```
Comment (cloud-fan) [inline python/pyspark/sql/connect/client.py:166]: I think the returned schema should always be a StructType?

Comment (cloud-fan) [reply python/pyspark/sql/connect/client.py:166]: nvm, this is called by `schema`

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala (hunk 2 above)
Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala:69]: Entire nullability is ignored so far in current Connect implementation.

Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala]: Done. I set `nullablity` now.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**